在原课程中详细介绍了理论概念与底层代码，然而在实践中还是要懂得如何使用一些机器学习框架如sklearn快速准确实现目的，本讲义以波士顿房价预测为例，展示用sklearn机器学习框架如何实现线性回归

In [2]:
from sklearn.datasets import load_boston   # 导入数据集
import matplotlib.pyplot as plt            #  绘图工具

In [3]:
dataset = load_boston()     # 加载数据
x_data = dataset.data       # 导入所有特征变量
y_data = dataset.target     # 导入target（房价）
feature_names = dataset.feature_names #导入特征名

In [19]:
print(dataset.DESCR) # 在此可以查看特征变量含义

.. _boston_dataset:

Boston house prices dataset
---------------------------

**Data Set Characteristics:**  

    :Number of Instances: 506 

    :Number of Attributes: 13 numeric/categorical predictive. Median Value (attribute 14) is usually the target.

    :Attribute Information (in order):
        - CRIM     per capita crime rate by town
        - ZN       proportion of residential land zoned for lots over 25,000 sq.ft.
        - INDUS    proportion of non-retail business acres per town
        - CHAS     Charles River dummy variable (= 1 if tract bounds river; 0 otherwise)
        - NOX      nitric oxides concentration (parts per 10 million)
        - RM       average number of rooms per dwelling
        - AGE      proportion of owner-occupied units built prior to 1940
        - DIS      weighted distances to five Boston employment centres
        - RAD      index of accessibility to radial highways
        - TAX      full-value property-tax rate per $10,000
        - PTRATIO  pu

In [10]:
from sklearn.model_selection import train_test_split  # 导入切分数据集的工具
X_train,X_test,y_train,y_test = train_test_split(x_data,y_data,random_state=0,test_size=0.2) # 划分数据集

In [17]:
from sklearn.linear_model import LinearRegression  # 导入线性回归工具库
lr = LinearRegression()  # 创建模型实例
lr.fit(X_train,y_train)  # 使用训练数据进行参数估计
predict = lr.predict(X_test) # 回归预测

In [18]:
# 查看模型拟合情况
from sklearn.metrics import r2_score #导入模型评价工具
score = r2_score(y_test, predict)    #根据预测结果和实际标签进行模型评价 
print('模型拟合优度为：{:.2f}'.format(score))

模型拟合优度为：0.59


回归模型的拟合优度(也叫R方)常被用来检验回归模型的拟合效果，区间为[0,1]， 值越高代表拟合得越好，该模型没有经过任何特征处理，已经取得0.59的得分，但是通过后续学习的特征处理等方法，可以有效提高它的分数，这也是入门机器学习者和专业机器学习从业者的重要分水岭。